apollo-composition/src/lib.rs

```diff
@@ -101,6 +116,87 @@ pub trait HybridComposition {
             return;
         };
 
+        // Overrides mess with the supergraph in ways that can be difficult to detect when
+        // expanding connectors; the supergraph may omit overridden fields and other shenanigans.
+        // To allow for a better developer experience, we check here if any connector-enabled subgraphs
+        // have fields overridden and error early.
+        let validations_by_subgraph_name =
+            HashMap::<_, _>::from_iter(validation_results.into_iter());
+        let mut override_errors = Vec::new();
+        for (subgraph_name, ValidationResult { schema, .. }) in validations_by_subgraph_name.iter()
+        {
+            // We need to grab all fields in the schema since only fields can have the @override
+            // directive attached
+            macro_rules! extract_directives {
+                ($node:ident) => {
+                    $node
+                        .fields
+                        .iter()
+                        .flat_map(|(name, field)| {
+                            field
+                                .directives
+                                .iter()
+                                .map(move |d| (format!("{}.{}", $node.name, name), d))
+                        })
+                        .collect::<Vec<_>>()
+                };
+            }
+            let override_directives = schema
+                .types
+                .values()
+                .flat_map(|v| match v {
+                    ExtendedType::Object(node) => extract_directives!(node),
+                    ExtendedType::Interface(node) => extract_directives!(node),
+                    ExtendedType::InputObject(node) => extract_directives!(node),
+
+                    // These types do not have fields
+                    ExtendedType::Scalar(_) | ExtendedType::Union(_) | ExtendedType::Enum(_) => {
+                        Vec::new()
+                    }
+                })
+                .filter(|(_, directive)| directive.name == "override");
```

Comment:
unfortunately, we don't know if the directive name is `@override`, `@federation__override`, or some other alias the user chose.

let's check for `directive.name == "override" || directive.name == "federation__override"` and put a TODO for handling aliases later.